tests/integration/test_setflags.py

```diff
@@ -0,0 +1,146 @@
+# Copyright 2021-2022 NVIDIA Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import numpy as np
+import pytest
+from legate.core import LEGATE_MAX_DIM
+
+import cunumeric as num
+
+
+@pytest.mark.parametrize("write", (None, False, True, 1, -1, 100, "11"))
+def test_set_value(write):
+    array_np = np.array(3)
+    array_num = num.array(3)
+    assert array_np.flags == array_num.flags
+    array_np.setflags(write)
+    array_num.setflags(write)
+    assert array_np.flags == array_num.flags
+
+
+@pytest.mark.xfail
+def test_array_default_flags():
+    array_np = np.array([0, 0, 0, 0, 0])
+    array_num = num.array([0, 0, 0, 0, 0].copy())
+    assert array_np.flags == array_num.flags
+    # array_np.flags
+    #          C_CONTIGUOUS : True
+    #          F_CONTIGUOUS : True
+    #          OWNDATA : True
+    #          WRITEABLE : True
+    #          ALIGNED : True
+    #          WRITEBACKIFCOPY : False
+    # array_num.flags
+    #          C_CONTIGUOUS : True
+    #          F_CONTIGUOUS : True
+    #          OWNDATA : False
```

Comment:
> we are always going to have this difference for owndata, is that right?

Yes, `owndata` will generally be `False` when querying it on a cuNumeric array. That's because these flags reflect the characteristics of the NumPy array that will get created from the cuNumeric array if the user code does a conversion. And because the converted NumPy array shares the data of the original cuNumeric array, it will always have `owndata == False`.

The one exception is if you do something like:

```
>>> import cunumeric as num
>>> import numpy as np
>>> np_arr = np.ones((4,))
>>> num_arr = num.asarray(np_arr)
>>> num_arr.flags
  C_CONTIGUOUS : True
  F_CONTIGUOUS : True
  OWNDATA : True
  WRITEABLE : True
  ALIGNED : True
  WRITEBACKIFCOPY : False
```

As before, the `flags` on the cuNumeric array represent the flags on the NumPy array that will get generated if the user code does a conversion. However, in this case the NumPy-equivalent array you get back is the same as the one originally used to create the cuNumeric array, which did own its data.